[Bug] v.hull fails to compile in GRASS 7.8.5

**Describe the bug**
When compiling GRASS 7.8.4 against GDAL 3.3.0 the v.hull module fails.

**To Reproduce**
```
GRASS GIS 7.8.5 exported compilation log
--------------------------------------------------
Started compilation: Mon 03 May 2021 10:51:18 PM MST
--
Errors in:
/tmp/makepkg/grass/src/grass-7.8.5/vector/v.hull
```

and the error from the v.hull folder in the src package gives:

```
grass/src/grass-7.8.5/vector/v.hull % make
gcc -D_FORTIFY_SOURCE=2 -march=native -O2 -pipe -fstack-protector-strong -fno-plt   -I/tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include -I/tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include   -I/usr/include -I/usr/include -DPACKAGE=\""grassmods"\"   -I/tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include -I/tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include -DRELDIR=\"vector/v.hull\" -o OBJ.x86_64-pc-linux-gnu/chull.o -c chull.c
In file included from /usr/include/ogr_api.h:47,
                 from /tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include/grass/vect/dig_structs.h:27,
                 from /tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include/grass/vect/digit.h:3,
                 from /tmp/makepkg/grass/src/grass-7.8.5/dist.x86_64-pc-linux-gnu/include/grass/vector.h:4,
                 from chull.c:27:
chull.c:34:19: error: expected ‘;’, identifier or ‘(’ before ‘_Bool’
   34 | { BFALSE, BTRUE } bool;
      |                   ^~~~
make: *** [../../include/Make/Compile.make:32: OBJ.x86_64-pc-linux-gnu/chull.o] Error 1
```

**System description (please complete the following information):**

- Operating System: Arch Linux (5.11.6 kernel)
- GRASS GIS 7.8.4
- GDAL 3.3.0

**Additional context**
Downgrading GDAL to 3.2.2 lets GRASS 7.8.4 compile.

--- a/include/grass/gis.h
+++ b/include/grass/gis.h
@@ -23,6 +23,8 @@
 /* System include files */
 #include <stdio.h>
 #include <stdarg.h>
+#include <stdbool.h>
+
 
 /* Grass and local include files */
 #include <grass/config.h>
@@ -51,13 +53,14 @@ static const char *GRASS_copyright __attribute__ ((unused))
 #define G_gisinit(pgm) G__gisinit(GIS_H_VERSION, (pgm))
 #define G_no_gisinit() G__no_gisinit(GIS_H_VERSION)
 
-/* Define TRUE and FALSE for boolean comparisons */
+/* For boolean values and comparisons use the C99 type 'bool' with values 'true' */
+/* and 'false' For historical reasons 'TRUE' and 'FALSE' are still valid.        */
 #ifndef TRUE
-#define TRUE 1
+#define TRUE true
 #endif
 
 #ifndef FALSE
-#define FALSE 0
+#define FALSE false
 #endif
 
 #if (defined(_FILE_OFFSET_BITS) && _FILE_OFFSET_BITS == 64) || (__APPLE__ && __LP64__)

--- a/lib/lidar/lidar.h
+++ b/lib/lidar/lidar.h
@@ -62,9 +62,6 @@
     /* INTERPOLATOR */
 #define P_BILINEAR 		1
 #define P_BICUBIC 		0
-    /* Boolean definitions */
-#define TRUE 			1
-#define FALSE 			0
 
 /*----------------------------------------------------------------------------------------------------------*/
     /*STRUCTS DECLARATION */

--- a/raster/r.param.scale/param.h
+++ b/raster/r.param.scale/param.h
@@ -18,8 +18,6 @@
 				/* 'blank' edge around raster.          */
 #define MAX_WSIZE 499		/* Maximum dimensions of window.        */
 				/* Some useful labels.                  */
-#define TRUE 1
-#define FALSE 0
 
 #define RAD2DEG M_R2D
 #define DEG2RAD M_D2R

--- a/raster/r.surf.idw/main.h
+++ b/raster/r.surf.idw/main.h
@@ -1,8 +1,6 @@
 #include <grass/raster.h>
 
 #define         SHORT           short
-#define         TRUE    1
-#define         FALSE   0
 
 #define MELEMENT        struct Melement
 MELEMENT {

--- a/raster3d/r3.showdspf/Ball.c
+++ b/raster3d/r3.showdspf/Ball.c
@@ -7,8 +7,6 @@
 #include "Ball.h"
 #include "BallMath.h"
 #include <stdio.h>
-#define TRUE 1
-#define FALSE 0
 
 HMatrix mId = { {1, 0, 0, 0}
 , {0, 1, 0, 0}

--- a/raster3d/r3.showdspf/togif.c
+++ b/raster3d/r3.showdspf/togif.c
@@ -324,9 +324,6 @@ static int ditherrow(unsigned short *r, unsigned short *g, unsigned short *b,
  *
  *****************************************************************************/
 
-#define TRUE 1
-#define FALSE 0
-
 
 /************************** BumpPixel() ********************************/
 /*

--- a/vector/v.hull/chull.c
+++ b/vector/v.hull/chull.c
@@ -22,16 +22,13 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
+#include <stdbool.h>
 
 #include <grass/gis.h>
 #include <grass/vector.h>
 #include <grass/glocale.h>
 
 #include "globals.h"
-
-/*Define Boolean type */
-typedef enum
-{ BFALSE, BTRUE } bool;
 
 /* Define vertex indices. */
 #define X   0
@@ -76,10 +73,10 @@ struct tFaceStructure
 };
 
 /* Define flags */
-#define ONHULL   	BTRUE
-#define REMOVED  	BTRUE
-#define VISIBLE  	BTRUE
-#define PROCESSED	BTRUE
+#define ONHULL   	true
+#define REMOVED  	true
+#define VISIBLE  	true
+#define PROCESSED	true
 
 /* Global variable definitions */
 tVertex vertices = NULL;
@@ -436,7 +433,7 @@ bool AddOne(tVertex p)
     tFace f;
     tEdge e, temp;
     long int vol;
-    bool vis = BFALSE;
+    bool vis = false;
 
 
     /* Mark faces visible from p. */
@@ -446,7 +443,7 @@ bool AddOne(tVertex p)
 
 	if (vol < 0) {
 	    f->visible = VISIBLE;
-	    vis = BTRUE;
+	    vis = true;
 	}
 	f = f->next;
     } while (f != faces);
@@ -454,7 +451,7 @@ bool AddOne(tVertex p)
     /* If no faces are visible from p, then p is inside the hull. */
     if (!vis) {
 	p->onhull = !ONHULL;
-	return BFALSE;
+	return false;
     }
 
     /* Mark edges in interior of visible region for deletion.
@@ -470,7 +467,7 @@ bool AddOne(tVertex p)
 	    e->newface = MakeConeFace(e, p);
 	e = temp;
     } while (e != edges);
-    return BTRUE;
+    return true;
 }
 
 /*---------------------------------------------------------------------

--- a/vector/v.hull/chull.c
+++ b/vector/v.hull/chull.c
@@ -22,7 +22,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include <stdbool.h>
 
 #include <grass/gis.h>
 #include <grass/vector.h>

